cluster-autoscaler/processors/processors.go

```diff
@@ -0,0 +1,45 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package processors
+
+import (
+	"k8s.io/autoscaler/cluster-autoscaler/processors/nodegroups"
+	"k8s.io/autoscaler/cluster-autoscaler/processors/pods"
+)
+
+// AutoscalingProcessors are a set of customizable processors used for encapsulating
+// various heuristics used throught Cluster Autoscaler code.
+type AutoscalingProcessors struct {
+	// PodListProcessor is used to process list of unschedulable pods before autoscaling.
+	PodListProcessor pods.PodListProcessor
+	// NodeGroupListProcessor is used to process list of NodeGroups that can be used in scale-up.
+	NodeGroupListProcessor nodegroups.NodeGroupListProcessor
+}
+
+// DefaultProcessors returns default set of processors.
+func DefaultProcessors() *AutoscalingProcessors {
+	return &AutoscalingProcessors{
+		PodListProcessor:       pods.NewDefaultPodListProcessor(),
+		NodeGroupListProcessor: nodegroups.NewDefaultNodeGroupListProcessor()}
+}
+
+// TestProcessors returns a set of simple processors for use in tests.
+func TestProcessors() *AutoscalingProcessors {
```

Comment:
As commented previously it would be nice to get rid of this function and handle the defaults safely.

It is also slightly confusing that NoOpPodListProcessor is created both here and in DefaultProcessors() but in two different ways.

Reply:
I expect the default implementations to be non-empty for most processors (in particular I still hope to move FilterOutSchedulable to PodListProcessor). And I expect more processors to be added very soon. So I think it makes sense to add a dedicated set of no-op processors for tests (we may change them to mocks if we ever need to).